File: MiniCube.cpp
// MiniCube.cpp
#include "MiniCube.h"
#include <cstring>

MiniCube::MiniCube() : left(0), right(0), mid(0) {
    strcpy(this->name, "");
}

MiniCube::MiniCube(char n[4], int l, int r, int t) : left(l), right(r), mid(t) {
    strcpy(this->name, n);
}

MiniCube MiniCube::clone() const {
    return MiniCube((char*)name, left, right, mid);
}

File: MiniCube.js
class MiniCube {
    constructor(name, left, right, mid) {
        this.name = name;
        this._left = left;
        this._right = right;
        this._mid = mid;
    }

    get left() {
        return this._left;
    }

    set left(value) {
        this._left = value;
    }

    get right() {
        return this._right;
    }

    set right(value) {
        this._right = value;
    }

    get mid() {
        return this._mid;
    }

    set mid(value) {
        this._mid = value;
    }
}

export default MiniCube;